### PR TITLE
net: Restore default whitelistrelay to true

### PR DESCRIPTION
**Pull request by NicolasDorier**
I thought `whitelistrelay` default was `false` when it is `true`.

The root of the issue come from the fact that all references to `DEFAULT_` are not in the scope of this file, so hard coding of default values are used everywhere in `net.cpp`. I think that in a separate PR we should fix that more fundamentally everywhere.

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -911,7 +911,7 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     if (NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_ISIMPLICIT)) {
         NetPermissions::ClearFlag(permissionFlags, PF_ISIMPLICIT);
         if (gArgs.GetBoolArg("-whitelistforcerelay", false)) NetPermissions::AddFlag(permissionFlags, PF_FORCERELAY);
-        if (gArgs.GetBoolArg("-whitelistrelay", false)) NetPermissions::AddFlag(permissionFlags, PF_RELAY);
+        if (gArgs.GetBoolArg("-whitelistrelay", true)) NetPermissions::AddFlag(permissionFlags, PF_RELAY);
```
Comment (Sjors) [inline src/net.cpp]: Add comment `DEFAULT_WHITELISTRELAY = true`, so if anyone changes that default they won't miss this spot.

**Review by Sjors — APPROVED** [review]
utACK, agree this also needs a more structural solution.

**Comment by NicolasDorier** [comment]
Is there something strange with travis? all PRs seems not passing.

File: src/net.cpp [hunk 2]
```diff
@@ -910,8 +910,10 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     bool legacyWhitelisted = false;
     if (NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_ISIMPLICIT)) {
         NetPermissions::ClearFlag(permissionFlags, PF_ISIMPLICIT);
+        // DEFAULT_WHITELISTFORCERELAY = false
         if (gArgs.GetBoolArg("-whitelistforcerelay", false)) NetPermissions::AddFlag(permissionFlags, PF_FORCERELAY);
-        if (gArgs.GetBoolArg("-whitelistrelay", false)) NetPermissions::AddFlag(permissionFlags, PF_RELAY);
+        // DEFAULT_WHITELISTRELAY = true
```
Comment (MarcoFalke) [inline src/net.cpp]: What is this?

`DEFAULT_WHITELISTRELAY` has nothing to do with validity of blocks or transactions. Please just move it from validation to net or something

File: test/functional/p2p_permissions.py [hunk 3]
```diff
@@ -22,12 +22,19 @@ def set_test_params(self):
...
-        ["-whitelist=127.0.0.1", "-whitelistrelay"],
+        # default permissions (no specific permissions)
+        ["-whitelist=127.0.0.1"],
         ["relay", "noban", "mempool"],
         True)
 
+        self.checkpermission(
+        # relay permission removed (no specific permissions)
+        ["-whitelist=127.0.0.1", "-whitelistrelay=0"],
+        ["noban", "mempool"],
```
Comment (MarcoFalke) [inline test/functional/p2p_permissions.py]: Generally we intend with 4 spaces after an opening `(` that is immediately followed by a new line

Comment (NicolasDorier) [reply test/functional/p2p_permissions.py]: you mean that you want me to move the comment above?

File: src/net.cpp (hunk 1 above)
Comment (NicolasDorier) [reply src/net.cpp]: Actually I can easily move the constant (https://github.com/bitcoin/bitcoin/pull/16631#discussion_r314708245) testing it.

**Comment by emilengler** [comment]
**NACK** sorry, I think removing args shouldn't be done instantly. They should go through a depraction process and be removed after one major version has been released. Otherwise it could break compatibility.

**Comment by NicolasDorier** [comment]
@emilengler this PR is precisely about restoring compatibility that I broke during previous PR.

**Comment by emilengler** [comment]
Concept ACK

**Comment by NicolasDorier** [comment]
As @MarcoFalke suggested, I moved `DEFAULT_WHITELISTRELAY/DEFAULT_WHITELISTFORCERELAY` to `net.h`. It build fine, and is a more natural place for those constant to be.

I also reformatted the `p2p_permissions.py` file (indent after `(\n` opening )

**Comment by Sjors** [comment]
ACK 2e35132c084c812a536d2738ce1993e79112908d

> Is there something strange with travis? all PRs seems not passing.

Yes, one more rebase should make that go away, see #16633.

**Comment by NicolasDorier** [comment]
@Sjors oops, fixed the chmod, sorry for that.

**Comment by NicolasDorier** [comment]
Fixing the chmods on https://github.com/bitcoin/bitcoin/pull/16635

**Comment by NicolasDorier** [comment]
I rebased and also fixed `rpc_setban`

**Comment by promag** [comment]
ACK 3b05f0f70fbaee5b5eaa0d1b6f3b9d32f44410bb.

nit, could split the moved code to a move-only commit.

**Comment by Sjors** [comment]
re-ACK 3b05f0f70fbaee5b5eaa0d1b6f3b9d32f44410bb

**Comment by laanwj** [comment]
ACK 3b05f0f70fbaee5b5eaa0d1b6f3b9d32f44410bb

**Pull request merged**